compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1065,6 +1065,19 @@ private boolean isErrorTypeAssignable(BErrorType source, BErrorType target, Set<
                 && target.typeIdSet.isAssignableFrom(source.typeIdSet);
     }
 
+    private boolean isErrorTypeAssignableWithNoTypeIds(BErrorType source, BErrorType target,
+                                                       Set<TypePair> unresolvedTypes) {
+        if (target == symTable.errorType) {
+            return true;
+        }
+        TypePair pair = new TypePair(source, target);
+        if (unresolvedTypes.contains(pair)) {
```

Comment:
```suggestion
        if (!unresolvedTypes.add(pair)) {
```
And we can remove `add` command below

Reply:
I believe this should prevent extra lookups or extra hash calculations. First to check if contains and second to add it again to the set.

Reply:
Changed